### ISSUE TITLE
FR: Ability to add custom domain names for VPN On Demand on iOS

### ISSUE THREAD

**Issue by michaelyork** [tailscale/tailscale#8915]
### What are you trying to do?

Nice to say today's VPN On Demand release! Would like to propose ability to add one's own domains to trigger Tailscale being enabled, versus the current implementation of `*.ts.net` only.

### How should we solve this?

Add one or more domains (including wildcards) of your own to trigger VPN On Demand.

**Comment by DentonGentry** [comment]
Could I ask what kinds of domains would be useful to add to that list?

**Comment by michaelyork** [comment]
@DentonGentry Sure! In my case, I've got many internal services behind a corporate domain (whose subdomains resolve to internal IP addresses that are only accessible via a Tailscale Subnet Router).

**Comment by DentonGentry** [comment]
Are those configured as SplitDNS domains in https://login.tailscale.com/admin/dns ?

I ask because we try to avoid adding extra configuration knobs, if the defaults can be made more applicable using configuration information which already exists.

**Comment by michaelyork** [comment]
Are you referring to Search Domains?

**Comment by DentonGentry** [comment]
Honestly if the domains appear in any of the features configured in https://login.tailscale.com/admin/dns, SplitDNS or Search Domains or otherwise, we could consider having the VPN-on-demand feature include those domains by default.

**Comment by michaelyork** [comment]
Absolutely! I personally use the domain as a Search Domain and I think that'd be a lovely solution. :)

**Comment by stevetrease** [comment]
I also use a domain name with split DNS and would welcome being able to add the split domain name to the list of VPN-on-demand domains in Tailscale.

**Comment by SebastianOsinski** [comment]
I don't use split DNS setting, however I have my own domain which has public DNS records set up to point to various Tailscale IP addresses. It would be great to be able to specify that domain (along with wildcards) and have Tailscale turn automatically when connecting to them. Or, alternatively, just have Tailscale turn automatically when phone tries to connect to `100.64.0.0/10` subnet

**Comment by andreyrd** [comment]
We have the same need (but using headscale) using both search domains and a custom base_domain configured for MagicDNS (which may be a headscale-only thing).
But as long as search domains are accounted for in the on demand matching - we're happy, since most of our internal services are on those.

**Comment by cmsj** [comment]
@SebastianOsinski I think it would make more sense to be strictly DNS based, because that is very unambiguous. Basing it off the subnet could lead to issues (e.g. your remote LAN is 192.168.0.0/24 and you're on a wifi network that uses 192.168.0.0/24)

**Comment by aptkingston** [comment]
Including SplitDNS domains or search domains would be a fantastic feature. Personally I'd still agree that a totally flexible list of domains would be even better - but I understand you want to avoid cluttering the UI, so as long as there is *some* way to include other domains than `*.ts.net` as VPN on demand domains then that would be a huge win.

I also publish public DNS records pointing to Tailscale IP addresses, and host multiple services as subdomains (`foo.vpn.domain.com`, `bar.vpn.domain.com`) which makes it incompatible with normal magic DNS. I'd really love to take advantage of VPN on demand to minimise battery drain.

**Comment by MarcelMertens** [comment]
Hi, i have the same request.
I'm using Tailscale to connect to Homeassistant via an internal DNS name resolved via my internal DNS server.
So VPN On Demand via Split-DNS / Search-DNS would be a great addition. Or maybe add OnDemand option for selected Apps (Apple iOS).

Kind Regards

**Comment by xshro** [comment]
Hello, It will be very useful this future for ios devices battery. ❎❎❎

**Comment by RefineryX** [comment]
I was about to submit the same request, I’m mainly use Tailscale to connect back to my Home Assistant. As such, I have to constantly enable and disable Tailscale from my iPhone.

**Comment by Simie** [comment]
+1 from me, I’m using my own domain and dns for my local network’s services and integrated tailscale seamlessly via Search Domains. Everything works great when on cellular/non-wifi when tailscale is connected, but it drains my battery pretty fast.

**Comment by whalecoiner** [comment]
> Everything works great when on cellular/non-wifi when tailscale is connected, but it drains my battery pretty fast.

Yeah as my iPhone gets older and the battery capacity dies this becomes more and more of an issue!

**Comment by ryansouza** [comment]
I have not been able to get on-demand to work from my phone at all but in theory would this work by CNAME from a personal domain to the tailnet host? Would the phone’s resolver trigger the on-demand when it goss to recursively resolve `something.ts.net` to an ip?

**Comment by isvein** [comment]
I too use custom domain and as it is now I need to have TS on all the time.

**Comment by hardtimes** [comment]
I too would like this to access my internal domain on request, rather than having to open and close the vpn connection all the time from iOS.

**Comment by HackXIt** [comment]
I too am interested in this feature.

**Comment by vivekmumbles** [comment]
Any update on this feature?
